sdk/core/azure-core/src/datetime.cpp

```diff
@@ -863,6 +863,18 @@ std::string DateTime::ToString(DateFormat format, TimeFractionFormat fractionFor
         "Unrecognized date format (" + std::to_string(static_cast<int64_t>(format)) + ").");
   }
 
+  switch (fractionFormat)
+  {
+    case TimeFractionFormat::DropTrailingZeros:
+    case TimeFractionFormat::AllDigits:
+    case TimeFractionFormat::Truncate:
+      break;
+    default:
+      throw std::invalid_argument(
+          "Unrecognized time fraction format ("
+          + std::to_string(static_cast<int64_t>(fractionFormat)) + ").");
```

Comment:
Thank you Anton - that was going to be my next suggestion once we got past the "use a switch statement rather than a set of if/then statements" suggestion :).

Reply:
Having said that, the static cast to int64_t has the side effect of sign extending the underlying type to help catch arithmetic errors.

Or you could just let the compiler do the work with a switch statement :).